### PR TITLE
[incubator-kie-issues-1195] Process Variable support multiple metadata fields

### PR DESCRIPTION
**Pull request by elguardian**
Closes: https://github.com/apache/incubator-kie-issues/issues/1195

### CODE REVIEW

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/VariableTagsTest.java [hunk 1]
```diff
@@ -41,6 +47,19 @@
...
+    @Test
+    public void testVariableMultipleMetadata() throws Exception {
+        Application app = ProcessTestHelper.newApplication();
+        ApprovalWithRequiredVariableTagsProcess processDefinition = (ApprovalWithRequiredVariableTagsProcess) ApprovalWithRequiredVariableTagsProcess.newProcess(app);
+
+        RuleFlowProcess processEngine = (RuleFlowProcess) processDefinition.process();
+        Optional<Variable> var = processEngine.getVariableScope().getVariables().stream().filter(e -> "approver".equals(e.getName())).findAny();
+        assertThat(var).isPresent();
+        assertThat(var.get().getMetaData()).hasSize(3);
+        assertThat(var.get().getMetaData()).flatExtracting("ItemSubjectRef", "approver", "customTags");
```
Comment (martinweiler) [inline jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/VariableTagsTest.java:59]: Why is "ItemSubjectRef" and "approver" also considered metadata?

Should there be a check for the metadata value as well, eg. 
```
assertThat(var.get().getMetaData().get("customTags")).isEqualTo("required");
```

**Comment by kie-ci3** [comment]
**PR job** `#3` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-kogito-pipelines/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-kogito-runtimes -u https://github.com/apache/incubator-kie-kogito-runtimes/pull/3508 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3508/3/display/redirect

**Test results:**
- PASSED: 3267
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3508/3/testReport/org.kie.kogito.mongodb.transaction/AbstractTransactionManagerIT/testAbort/">org.kie.kogito.mongodb.transaction.AbstractTransactionManagerIT.testAbort</a></summary>
Command failed with error 246 (SnapshotUnavailable): 'Unable to read from a snapshot due to pending collection catalog changes; please retry the operation. Snapshot timestamp is Timestamp(1715685325, 0). Collection minimum is Timestamp(1715685325, 2)' on server 172.18.0.1:32799. The full response is {"errorLabels": ["TransientTransactionError"], "operationTime": {"$timestamp": {"t": 1715685324, "i": 6}}, "ok": 0.0, "errmsg": "Unable to read from a snapshot due to pending collection catalog changes; please retry the operation. Snapshot timestamp is Timestamp(1715685325, 0). Collection minimum is Timestamp(1715685325, 2)", "code": 246, "codeName": "SnapshotUnavailable", "$clusterTime": {"clusterTime": {"$timestamp": {"t": 1715685325, "i": 3}}, "signature": {"hash": {"$binary": {"base64": "AAAAAAAAAAAAAAAAAAAAAAAAAAA=", "subType": "00"}}, "keyId": 0}}}
</details>

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/VariableTagsTest.java (hunk 1 above)
Comment (elguardian) [reply jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/VariableTagsTest.java]: done, just checked all entries

**Review by martinweiler — APPROVED** [review]
LGTM @elguardian

**Review by pefernan — APPROVED** [review]

**Pull request merged**